controllers/slicegateway/slicegateway.go

```diff
@@ -1534,3 +1547,43 @@ func (r *SliceGwReconciler) ReconcileIntermediateGatewayDeployments(ctx context.
 
 	return ctrl.Result{}, nil, false
 }
+
+// createPodDisruptionBudgetForSliceGatewayPods checks for PodDisruptionBudget objects in the cluster that match the
+// slice gateway pods, and if missing, it creates a PDB with minimum availability of 1 so at least one pod remains in
+// case of a disruption.
+func (r *SliceGwReconciler) createPodDisruptionBudgetForSliceGatewayPods(ctx context.Context, log logr.Logger,
+	client client.Client, sliceName string, sliceGateway *kubeslicev1beta1.SliceGateway) error {
+	log = log.WithValues("sliceName", sliceName, "sliceGwName", sliceGateway.Name)
+
+	// List PDBs in cluster that match the slice gateway pods
+	pdbs, err := listPodDisruptionBudgetForSliceGateway(ctx, client, sliceName, sliceGateway.Name)
+	if err != nil && !apierrors.IsNotFound(err) {
+		// When some other error occurred, return to make the current request requeued for reconciler to reattempt.
+		return err
+	}
+
+	// Check if PDB already exists for the slice  gateway pods
+	if len(pdbs) > 0 {
+		log.Info("PodDisruptionBudget matching the slice and slice gateway already exists. Skipping creation.")
+
+		return nil
+	}
+
+	// Create PDB manifest with minimum availability of 1 pod
+	pdb := formatPodDisruptionBudget(sliceName, sliceGateway.Name, intstr.FromInt(1))
+
+	// Set SliceGateway instance as the owner and controller for PDB
+	ctrl.SetControllerReference(sliceGateway, pdb, r.Scheme)
+
+	// Create PDB for slice gateway's pod to at least have 1 instance of pods on
+	// each worker when disruption has occurred
+	err = client.Create(ctx, pdb)
+	if err != nil && !apierrors.IsAlreadyExists(err) {
```

Comment:
What is there is some other error, not resource already exist. We are not handling that and returning success instead

Reply:
@YachikaRalhan Updated as per our discussion. Please review. Thanks!